naga/tests/out/msl/atomicTexture-int64.msl

```diff
@@ -0,0 +1,18 @@
+// language: metal3.1
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using metal::uint;
+
+
+struct cs_mainInput {
+};
+kernel void cs_main(
+  metal::uint3 id [[thread_position_in_threadgroup]]
+, metal::texture2d<uint, metal::access::write> image [[user(fake0)]]
+) {
+    atomic_max_explicit(&image[metal::uint2(metal::int2(0, 0))], 1uL, metal::memory_order_relaxed);
```

Comment:
error: type 'metal::texture2d<uint, metal::access::write>' (aka 'texture2d<unsigned int, metal::access::write>') does not provide a subscript operator
idk how to get a pointer to a texel on metal, and https://github.com/KhronosGroup/SPIRV-Cross/issues/835 does not inspire hope

Reply:
Based on the spec, I believe it should be `image.atomic_max(coord, value);`
![image](https://github.com/user-attachments/assets/e71cb8e7-b6e9-43e5-baf3-722c4fd7ec38)

Reply:
i get now
```
error: no member named 'atomic_max' in 'metal::texture2d<unsigned int, metal::access::write>'
    image.atomic_max(metal::uint2(metal::int2(0, 0)), 1uL, metal::memory_order_relaxed);
    ~~~~~ ^
```

i think 64 bit atomics have different syntax than the 32 bit, https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf page 251
<img width="599" alt="{5691D177-355B-40C8-927A-CFDA2C4B6300}" src="https://github.com/user-attachments/assets/39f6457b-05ae-4655-8cf7-677d1292add6">

Reply:
Those are for buffers I believe, not textures. Notice how it says Metal 2.4 and not 3.1.

You want atomic_max based on this section:
![image](https://github.com/user-attachments/assets/f5a5c6e9-0eab-45cd-bc68-15543625296b)

Reply:
im following this to the letter and still erroring
```
error: no member named 'atomic_max' in 'metal::texture2d<unsigned long, metal::access::write>'
    image.atomic_max(metal::uint2(metal::int2(0, 0)), 1uL);
    ~~~~~ ^
```